test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetSignCommandTest.cs

```diff
@@ -0,0 +1,523 @@
+            var packagePath = @"foo\bar.nupkg";
+            var timestamper = "http://foo.bar";
+            var certificatePath = @"\\foo\bar.pfx";
+            var timestampHashAlgorithm = "MD5";
+            var mockConsole = new Mock<IConsole>();
+            var signCommand = new SignCommand
+            {
+                Console = mockConsole.Object,
+                Timestamper = timestamper,
+                CertificatePath = certificatePath,
+                TimestampHashAlgorithm = timestampHashAlgorithm
+            };
+
+            signCommand.Arguments.Add(packagePath);
+
+            // Act & Assert
+            var ex = Assert.Throws<ArgumentException>(() => signCommand.GetSignArgs());
+            Assert.Equal(string.Format(_invalidArgException, nameof(signCommand.TimestampHashAlgorithm)), ex.Message);
+        }
+
+        [Fact]
+        public void SignCommandArgParsing_ValidArgs_CertFingerprintAsync()
+        {
+            //Debugger.Launch();
+            // Arrange
+            var packagePath = @"foo\bar.nupkg";
+            var timestamper = "http://foo.bar";
+            var certificateFingerprint = new Guid().ToString();
+            var hashAlgorithm = "sha256";
+            Enum.TryParse(hashAlgorithm, ignoreCase: true, result: out HashAlgorithmName parsedHashAlgorithm);
+            var timestampHashAlgorithm = "sha512";
+            Enum.TryParse(timestampHashAlgorithm, ignoreCase: true, result: out HashAlgorithmName parsedTimestampHashAlgorithm);
+            var storeName = "CertificateAuthority";
+            Enum.TryParse(storeName, ignoreCase: true, result: out StoreName parsedStoreName);
+            var storeLocation = "LocalMachine";
+            Enum.TryParse(storeLocation, ignoreCase: true, result: out StoreLocation parsedStoreLocation);
+            var nonInteractive = true;
+            var overwrite = true;
+            var outputDir = @".\bar";
+            var csp = "csp_name";
+            var kc = "kc_name";
+            var mockConsole = new Mock<IConsole>();
+            mockConsole.Setup(c => c.Verbosity).Returns(Verbosity.Detailed);
+
+            var signCommand = new SignCommand
+            {
+                Console = mockConsole.Object,
+                Timestamper = timestamper,
+                CertificateFingerprint = certificateFingerprint,
+                CertificateStoreName = storeName,
+                CertificateStoreLocation = storeLocation,
+                HashAlgorithm = hashAlgorithm,
+                TimestampHashAlgorithm = timestampHashAlgorithm,
+                OutputDirectory = outputDir,
+                NonInteractive = nonInteractive,
+                Overwrite = overwrite,
+                CryptographicServiceProvider = csp,
+                KeyContainer = kc
+            };
+
+            signCommand.Arguments.Add(packagePath);
+
+            // Act & Assert
+            var signArgs = signCommand.GetSignArgs();
+
+            Assert.True(
```

Comment:
In this test and subsequent tests, would it be better to have multiple assertions?  If this assertion fails, you don't know the reason without debugging.  Bad!